scripts/make_tor_targets.py

```diff
@@ -0,0 +1,114 @@
+import json
+import hashlib
+
+from urllib.request import urlopen
+
+DIRAUTH_URL = "https://gitweb.torproject.org/tor.git/plain/src/app/config/auth_dirs.inc"
+BRIDGES_URL = "https://bridges.torproject.org/moat/circumvention/builtin"
+
+def parse_params(parts : list[str]) -> dict[str, str]:
+    params = {}
+    for p in parts:
+        k, v = p.split("=")
+        params[k] = [v]
+    return params
+
+def parse_bridge_line(line : str) -> tuple[str, dict]:
+    bridge = {}
+    parts = line.split(" ")
+    bridge["protocol"] = parts[0]
+    bridge["address"] = parts[1]
+    bridge["fingerprint"] = parts[2]
+    bridge["params"] = parse_params(parts[3:])
+
+    bridge_id = hashlib.sha256(
+            bridge["address"].encode("ascii") + bridge["fingerprint"].encode("utf-8")
+    ).hexdigest()
+
+    return bridge_id, bridge
+
+def get_bridges():
+    with urlopen(BRIDGES_URL) as resp:
+        j = json.loads(resp.read())
+
+    bridges = {}
+    for b in j["obfs4"]:
+        bridge_id, bd = parse_bridge_line(b)
+        assert bridge_id not in bridges
```

Comment:
You should probably explain with a comment why the policy is to fail the script in case of duplicates. (Other policies may be possible, such as using the first or the last entry, but here you chose explicitly to fail the script if that happens, which likely has some background reasoning. Because it may be that you're not the only future person running this script, I was wondering whether it would be best to explain the policy with a comment and provide additional instructions in the same comment regarding what the operator should do in this case.)

Reply:
It's failing because the data format is a dict, so if there is a duplicate key, you will end up overwriting the previous entry without noticing.

I agree this should be added as a comment. I will do so.